ui/src/CollectionExerciseDetails.js

```diff
@@ -240,6 +251,22 @@ class CollectionExerciseDetails extends Component {
       failedValidation = true;
     }
 
+    if (!this.state.newSmsUacQidMetadata.trim()) {
+      this.setState({ smsUacQidMetadataValidationError: true });
+      failedValidation = true;
+    } else {
+      try {
+        const parsedJson = JSON.parse(this.state.newSmsUacQidMetadata);
+        if (parsedJson.length === 0) {
```

Comment:
this length check assumes that the type is an array... which I don't think it should.

I think that the metadata should be just a JSON blob